docs/cyclefold_report.md

```diff
@@ -0,0 +1,211 @@
+   ```
+   Here:
+   - **p0 = (x_0, y_0)** and **p1 = (x_1, y_1)** are points on the second curve,
+   - **ℓ_0, ℓ_1** are scalars in the base field,
+   - **expected_x, expected_y** is the claimed result of **p0·ℓ_0 + p1·ℓ_1**.
+
+3. **Operation**: Inside the circuit, we:
+   - **Read** these eight values from the instance column.
+   - **Compute** **p0·ℓ_0 + p1·ℓ_1** in native field arithmetic (using the *Tiny Gate* approach).
+   - **Compare** the computed point with the "expected" coordinates **expected_x, expected_y**, enforcing equality constraints.
+
+4. **Tiny Gate**: The circuit uses a minimal, specialized gate 
+    ```
+    s[0] * s[1] * mul + s[0] * sum[0] + s[1] * sum[1] + rc - output = 0
+    ```
+    that can perform:
+   - Addition/subtraction in the base field
+   - Multiplication by a constant
+   - A few small subroutines (e.g., bit‐decomposition for scalars)
+   - A straightforward "selector" design to minimize overhead  
+   This yields a *very small* circuit, which is crucial to keep the second curve's cost down.
+
+5. **Folding**: After verifying the correct scalar multiplication, the **Support Circuit** is "folded" back into the main circuit via the **Sangria**/**Nova**‐like aggregator. This means the Step Folding Circuit only has to verify a handful of commitments and challenges, rather than re‐implementing all of the second curve's arithmetic.
+
+Thus, the Support Circuit *acts like a co‐processor* for cross‐curve operations, ensuring that the main circuit (on the first curve) remains simpler and more efficient, yet still fully accounts for delegated scalar multiplications.
+
+## 5. Benchmark
+
+### Hardware Configuration
+
+The experiments were performed on a high-end Linux server. The system's key hardware components are summarized below:
+
+- **System Overview**
+  - **System:** Super Server (OEM details not provided)
+  - **Bus:** H12SSL-i
+
+- **Memory**
+  - **BIOS Memory:** 64 KiB
+  - **Total System Memory:** 1 TiB
+  - **DIMM Modules:** 8 × 128 GiB DDR4 Synchronous Registered (Buffered) operating at 2667 MHz
+
+- **Processor & Cache**
+  - **Processor:** AMD EPYC 7702 64-Core Processor
+  - **Cache:**
+    - L1 Cache: 4 MiB
+    - L2 Cache: 32 MiB
+    - L3 Cache: 256 MiB
+
+### Trivial Circuit
+
+An identity circuit that simply passes inputs through unchanged. This circuit provides a baseline benchmark for measuring the pure overhead of the IVC infrastructures without any computational complexity.
+
+| Step        | Time in seconds |
+| ---         | ---             |
+| IVC::new    | 7.03            |
+| IVC::next   | 5.4             |
+| IVC::verify | 1.45            |
+
+### Poseidon Circuit
+
+A circuit implementing the Poseidon hash function with configurable parameters (POSEIDON_PERMUTATION_WIDTH=3, R_F1=4, R_P1=3). This benchmark evaluates how efficiently the IVC handle a common cryptographic primitive used in ZK applications.
+
+#### Simple hash function call
```

Comment:
is this data for cyclefold? how is it compared to sangaria fold?

Reply:
Yes, I planned to insert only Cyclefold data in all paragraphs except the last one, because otherwise you have to change the whole stand (because for a small number of gates the picture will be highly distorted)